classifier/__init__.py

```diff
@@ -1068,7 +1102,11 @@ def add(self, classification):
                     self.audience_weights[Classifier.AUDIENCE_YOUNG_ADULT] += (weight * 0.6)
                     self.audience_weights[Classifier.AUDIENCE_CHILDREN] += (weight * 0.4)
                     for audience in Classifier.AUDIENCES_ADULT:
-                        self.audience_weights[audience] -= weight * 0.5
+                        if audience != Classifier.AUDIENCE_ALL_AGES:
+                            # 'All Ages' is considered an adult audience,
+                            # but a generic 'juvenile' classification
+                            # is not evidence against it.
+                            self.audience_weights[audience] -= weight * 0.5
```

Comment:
This is tested by our change to test_juvenile_classification_is_split_between_children_and_ya.